src/MvvmNano.Forms/MvvmNanoFormsPresenter.cs

```diff
@@ -90,21 +95,29 @@ public void SetApplication(Application application)
                 .ToArray();
         }
 
+        bool IsMasterDetailApplication()
+        {
+            return Application is MvvmNanoMasterDetailApplication;
+        } 
+
+        public string GetViewNameByViewModel(Type viewModelType)
+        {
+            return viewModelType.Name.Replace(VIEW_MODEL_SUFFIX, VIEW_SUFFIX);
+        } 
+
         /// <summary>
         /// Navigates to a Page and automatically creates a new instance of
         /// the corresponding View Model. Also passes some parameters of the
         /// given type.
         /// </summary>
         public void NavigateToViewModel<TViewModel, TNavigationParameter>(TNavigationParameter parameter)
         {
-            var viewModel = CreateViewModel<TViewModel, TNavigationParameter>();
+            IViewModel<TNavigationParameter> viewModel = CreateViewModel<TViewModel, TNavigationParameter>();
             viewModel.Initialize(parameter);
-
-            IView view = CreateViewFor<TViewModel>();
-            view.SetViewModel(viewModel);
-
-            OpenPage(view as Page);
-        }
+            IView viewFor = CreateViewFor<TViewModel>();
+            viewFor.SetViewModel((IViewModel)viewModel);
+            StartOpeningPageAsync<TViewModel>(viewFor as Page);
```

Comment:
Btw.: I think I saw 2, 3 other places with such minor issues. Maybe you could take a look at the source with Jetbrains Rider or Visual Studio + ReSharper, which both provide much better code analysis than Xamarin Studio does.